### PR TITLE
Prevent empty paths to app.yaml and Dockerfile

### LINKED ISSUE #607: 'gcloud' is now auto-generating app.yaml and Dockerfile if not present - require them for custom deployments
Check it out. Run the 'gcloud app deploy' in a directory where there is only one war. If either one or both of them does not exist, it will create them.

We don't need to manually copy our templates.

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #607.

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java [hunk 1]
```diff
@@ -250,6 +250,12 @@ private void validateConfiguration() throws ConfigurationException {
           GctBundle.message("appengine.config.deployment.source.error"));
     } else if(versionOverrideCheckBox.isSelected() && StringUtils.isBlank(versionIdField.getText())) {
       throw new ConfigurationException(GctBundle.message("appengine.config.version.error"));
+    } else if (getConfigType() == ConfigType.CUSTOM) {
+      if (StringUtils.isBlank(appYamlPathField.getText()) ||
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java]: We can probably separate out these error messages to show one for a missing Dockerfile and one for a missing app.yaml.

Comment (chanseokoh) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java]: Expected that feedback. I'll do it.

File: google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties [hunk 2]
```diff
@@ -44,7 +44,8 @@ appengine.flex.config.generation.io.error=There was an error generating the {0}.
 appengine.config.project.panel.title=Project: {0}
 appengine.flex.config.user.specified.artifact.title=Select a JAR or WAR file
 appengine.flex.config.user.specified.artifact.error=Browse to a JAR or WAR file.
-appengine.flex.config.custom.deploy.title=App Engine Flexible Environment Configuration.
+appengine.flex.config.custom.deploy.title=App Engine Flexible Environment Configuration
+appengine.flex.config.custom.configfiles.error=Provide app.yaml and Dockerfile.
```
Comment (etanshaul) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: To keep it consistent with the jar/war error message, how about:

"Browse to an app.yaml file."
"Browse to a Dockerfile."

**Comment by etanshaul** [comment]
I'd also take a look at this closed PR: #555 

I made some changes there specifically to allow for empty Dockerfile / app.yaml to pass through to gcloud. Given that we are now requiring them, some of that logic might be able to be changed back (making the Dockerfile / app.yaml params @Nullable, for instance, may not make sense anymore).

**Comment by chanseokoh** [comment]
Not sure, but eventually we will allow leaving the fields blank and even remove the 'Generate...' buttons? In that case, it seems good to have the logic in #555. Otherwise, we may forget to do #555 again when we remove this validation.

**Comment by etanshaul** [comment]
I think that even if we later change it to allow blanks we should update the rest of the code to work with the current logic to keep things cohesive. 

Any time we modify that UI we should also be thinking about how it affects the deployment logic, but I take your point: maybe add a small comment in the UI code as a reminder?

**Comment by chanseokoh** [comment]
I looked at #555 more closely. I think it is necessary to restore the logic to check existence of user-provided paths. Currently, even if we add this validation, if a user enters a non-existing path, we silently ignore that error which will make gcloud auto-generate the files. I'll restore the logic.

**Comment by chanseokoh** [comment]
I reverted the changes in #555.

However, throwing RuntimeException does sound quite right (much like #605). There are other places where we throw RuntimeException, and I think they should be handled differently. There should be another PL for these changes.

**Comment by chanseokoh** [comment]
does sound --> doesn't sound

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineCloudType.java [hunk 3]
```diff
@@ -361,12 +361,12 @@ public void computeDeployments(@NotNull ComputeDeploymentsCallback callback) {
     public void disconnect() {
     }
 
-    @Nullable
+    @NotNull
     private File getFileFromFilePath(String filePath) {
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineCloudType.java]: I think there's some changes higher up in this class where this method is invoked that also need to be changed alongside this.

**Comment by etanshaul** [comment]
Yea - I think the RuntimeExceptions can be treated on a case by case basis. I'm not entirely sure if its appropriate here or not. I need to look at it more closely.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineCloudType.java (hunk 3 above)
Comment (chanseokoh) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineCloudType.java]: Thanks for catching that!

**Comment by etanshaul** [comment]
lgtm

**Pull request merged**